--- configure.ac.orig	2014-02-01 13:22:13.548268293 +0000
+++ configure.ac	2014-02-01 17:09:15.101315758 +0000
@@ -605,8 +605,15 @@
 	AS_HELP_STRING([--with-libdb=PREFIX],
 	[Prefix where libdb is installed]),
 	[
+		if test x$withval = xyes; then
+			withval=/usr
+		fi
 		DB_CFLAGS="-I$withval/include"
-		DB_LIBS="-L$withval/lib -ldb"
+		if test $withval = /usr; then
+			DB_LIBS="-L$withval/$libdir -ldb"
+		else
+			DB_LIBS="-L$withval/lib -ldb"
+		fi
 	],
 	[
 		if test -z "$DB_CFLAGS" -a -z "$DB_LIBS"; then
