--- configure.ac~	2013-12-04 15:08:42.000000000 +0000
+++ configure.ac	2013-12-18 13:03:21.625643833 +0000
@@ -545,9 +545,13 @@
 	AS_HELP_STRING([--with-libdb=PREFIX],
 	[Prefix where libdb is installed]),
 	[libdb_prefix="$withval"], [libdb_prefix='${prefix}'])
-
+if test x$libdb_prefix = xyes; then
+	libdb_prefix=/usr
+	DB_LIBS="-L$libdb_prefix/$libdir -ldb"
+else
+	DB_LIBS="-L$libdb_prefix/lib -ldb"
+fi
 DB_CFLAGS="-I$libdb_prefix/include"
-DB_LIBS="-L$libdb_prefix/lib -ldb"
 
 AC_MSG_CHECKING([Berkeley DB])
 save_cflags=$CFLAGS; CFLAGS=$DB_CFLAGS
