--- ./src/pages.c.orig	2016-10-29 11:48:04.819954478 +0100
+++ ./src/pages.c	2016-10-29 11:49:20.851010441 +0100
@@ -214,11 +214,14 @@
 	char buf[1];
 	ssize_t nread;
 
-	fd = open("/proc/sys/vm/overcommit_memory", O_RDONLY);
+	fd = syscall(SYS_open, "/proc/sys/vm/overcommit_memory", O_RDONLY);
 	if (fd == -1)
 		return (false); /* Error. */
 
-	nread = read(fd, &buf, sizeof(buf));
+	nread = syscall(SYS_read, fd, &buf, sizeof(buf));
+
+	syscall(SYS_close, fd);
+
 	if (nread < 1)
 		return (false); /* Error. */
 	/*
