--- src/lj_arch.h.orig	2015-04-14 18:16:22.928032393 +0000
+++ src/lj_arch.h	2015-04-14 18:17:20.395516753 +0000
@@ -37,7 +37,7 @@
 /* Select native target if no target defined. */
 #ifndef LUAJIT_TARGET
 
-#if defined(__i386) || defined(__i386__) || defined(_M_IX86)
+#if defined(__i386) || defined(__i386__) || defined(_M_IX86) || ( defined(__x86_64__) && defined(__ILP32__) )
 #define LUAJIT_TARGET	LUAJIT_ARCH_X86
 #elif defined(__x86_64__) || defined(__x86_64) || defined(_M_X64) || defined(_M_AMD64)
 #define LUAJIT_TARGET	LUAJIT_ARCH_X64
