--- ./src/lib/krb5/os/sendto_kdc.c~	2014-01-16 00:44:15.000000000 +0000
+++ ./src/lib/krb5/os/sendto_kdc.c	2014-02-01 11:27:33.693501195 +0000
@@ -372,7 +372,7 @@
     struct timeval tv;
 #endif
     krb5_error_code retval;
-    time_ms curtime, interval;
+    time_ms curtime = 0, interval;
 
     retval = get_curtime_ms(&curtime);
     if (retval != 0)
@@ -913,7 +913,7 @@
             void *msg_handler_data, struct conn_state **winner_out)
 {
     int e, selret = 0;
-    time_ms endtime;
+    time_ms endtime = 0;
     struct conn_state *state;
 
     *winner_out = NULL;
--- ./src/kadmin/dbutil/dump.c~	2014-01-16 00:44:15.000000000 +0000
+++ ./src/kadmin/dbutil/dump.c	2014-02-01 11:30:09.361751512 +0000
@@ -1487,7 +1487,8 @@
     kdb_log_context *log_ctx;
     krb5_boolean db_locked = FALSE, temp_db_created = FALSE;
     krb5_boolean verbose = FALSE, update = FALSE, iprop_load = FALSE;
-    uint32_t caller = FKCOMMAND, last_sno, last_seconds, last_useconds;
+    uint32_t caller = FKCOMMAND;
+    uint32_t last_sno = 0, last_seconds = 0, last_useconds = 0;
 
     /* Parse the arguments. */
     dbname = global_params.dbname;
