--- gdk/x11/gdkvisual-x11.c~	2014-12-01 21:37:31.000000000 +0000
+++ gdk/x11/gdkvisual-x11.c	2014-12-01 21:40:04.860791685 +0000
@@ -345,7 +345,7 @@
   /* If GL is available we want to pick better default/rgba visuals,
      as we care about glx details such as alpha/depth/stencil depth,
      stereo and double buffering */
-  _gdk_x11_screen_update_visuals_for_gl (screen);
+  //_gdk_x11_screen_update_visuals_for_gl (screen);
 }
 
 gint
