--- ./gdk/gdkdisplaymanager.c.orig	2014-11-30 19:01:11.687622360 +0000
+++ ./gdk/gdkdisplaymanager.c	2014-11-30 19:50:39.662994773 +0000
@@ -270,12 +270,12 @@
 #ifdef GDK_WINDOWING_WIN32
   { "win32",    _gdk_win32_display_open },
 #endif
-#ifdef GDK_WINDOWING_X11
-  { "x11",      _gdk_x11_display_open },
-#endif
 #ifdef GDK_WINDOWING_WAYLAND
   { "wayland",  _gdk_wayland_display_open },
 #endif
+#ifdef GDK_WINDOWING_X11
+  { "x11",      _gdk_x11_display_open },
+#endif
 #ifdef GDK_WINDOWING_MIR
   { "mir",  _gdk_mir_display_open },
 #endif
