--- libdbusmenu-gtk/Makefile.am~	2012-06-07 16:50:39.000000000 +0100
+++ libdbusmenu-gtk/Makefile.am	2013-09-23 12:22:13.004205746 +0100
@@ -63,6 +63,7 @@
 	-export-symbols-regex "^[^_].*"
 
 libdbusmenu_gtk_la_CFLAGS = \
+	$(DBUSMENUGLIB_CFLAGS) \
 	$(DBUSMENUGTK_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
@@ -70,7 +71,7 @@
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GTK\""
 
 libdbusmenu_gtk_la_LIBADD = \
-	$(top_builddir)/libdbusmenu-glib/libdbusmenu-glib.la \
+	$(DBUSMENUGLIB_LIBS) \
 	$(DBUSMENUGTK_LIBS)
 
 # We duplicate these here because Automake won't let us use $(VER) on the left hand side.
@@ -125,8 +126,7 @@
 	-I$(top_srcdir)
 DbusmenuGtk_0_4_gir_LDFLAGS = \
 	$(COVERAGE_LDFLAGS)
-DbusmenuGtk_0_4_gir_LIBS = libdbusmenu-gtk$(VER).la \
-	$(top_builddir)/libdbusmenu-glib/libdbusmenu-glib.la
+DbusmenuGtk_0_4_gir_LIBS = libdbusmenu-gtk$(VER).la
 DbusmenuGtk_0_4_gir_FILES = $(addprefix $(srcdir)/, $(introspection_sources))
 DbusmenuGtk_0_4_gir_NAMESPACE = DbusmenuGtk$(VER)
 DbusmenuGtk_0_4_gir_EXPORT_PACKAGES = dbusmenu-gtk$(VER)-0.4
--- configure.ac.orig	2013-09-23 12:42:39.634208950 +0100
+++ configure.ac	2013-09-23 12:44:54.914209304 +0100
@@ -36,7 +36,8 @@
 
 GLIB_REQUIRED_VERSION=2.26
 
-PKG_CHECK_MODULES(DBUSMENUGLIB, glib-2.0 >= $GLIB_REQUIRED_VERSION
+PKG_CHECK_MODULES(DBUSMENUGLIB, dbusmenu-glib-0.4 = $VERSION
+                                glib-2.0 >= $GLIB_REQUIRED_VERSION
                                 gio-2.0 >= $GLIB_REQUIRED_VERSION)
 
 AC_SUBST(DBUSMENUGLIB_CFLAGS)
--- Makefile.am~	2012-06-07 16:45:34.000000000 +0100
+++ Makefile.am	2013-09-23 12:47:08.435209653 +0100
@@ -14,7 +14,6 @@
 endif
 
 SUBDIRS = \
-	libdbusmenu-glib \
 	$(LIBDBUSMENUGTK_SUBDIR) \
 	tools \
 	$(TESTS_SUBDIR) \
